### ISSUE TITLE
reveal bug - reveal with big content, do not correctly positioning in small screen

### ISSUE THREAD

**Issue by remrub** [zurb/foundation-sites#7977]
Reveal with big content, do not correctly positioning in small screen. Imposible scroll to top. For example inaccessible Close button. 
Very please fix it. This is very important in my big project who have a lot modals.

1. Picture - Still good
![1 still good](https://cloud.githubusercontent.com/assets/13866469/12529910/fac8e430-c1d3-11e5-8923-e71d8424578c.JPG)

2. Picture - No good, no scrooll
![2 no good](https://cloud.githubusercontent.com/assets/13866469/12529911/face57e4-c1d3-11e5-953c-88e3ca4705ec.JPG)

**Comment by Peeeep** [comment]
I have the same issue... :/ As a temporary fix, you can add this line 
```javascript
//if smaller than medium, resize to 100% width minus any custom L/R margin
this.$element
  .css({
    'height': eleDims.windowDims.height, // add this line
    'width': eleDims.windowDims.width - (this.options.hOffset * 2)
  })
```
in your foundation.reveal.js around line 240 which will force a full screen (and scrollable) reveal modal on small screens. It's not ideal if the modal is tiny, but it should do for now.

**Comment by zurbchris** [comment]
If anyone effected by this bug can check out [this branch](https://github.com/zurb/foundation-sites/tree/update-reveal) and take a look at the new positioning algorithm, I would very much appreciate it. It's streamlined a toooon, and caches most of the values, so it doesn't recalc everything, every time the modal is opened if it doesn't need to.

**Comment by gakimball** [comment]
Fixed via 330018d577bfbdaf30dad96fa6a05e4a074419e6
